### PR TITLE
Construct a system from a subsystem

### LINKED ISSUE #1086: Support export of subsystem to a new system
Using subsystems capability to export a sub set of the data

### PR DESCRIPTION
**Pull request by daniel-thom**
Fixes #1086

### CODE REVIEW

File: src/base.jl [hunk 1]
```diff
@@ -2098,13 +2152,27 @@ function handle_component_removal!(sys::System, device::Device)
     return
 end
 
+function handle_component_removal!(sys::System, component::RegulationDevice)
```
Comment (daniel-thom) [inline src/base.jl:2155]: We had two existing bugs where we were not removing masked components from RegulationDevice or StaticInjectionSubsystem.

File: src/subsystems.jl [hunk 2]
```diff
@@ -240,8 +240,8 @@ function _check_topological_consistency(sys::System, component::Component)
                 sys,
                 component,
                 val,
-                "A component must be assigned to the same subsystems as its topological component(s).";
-                symmetric_diff = true,
+                "A component must be assigned to at least one subsystem as its topological component(s).";
```
Comment (daniel-thom) [inline src/subsystems.jl:243]: This check was not correct for a valid test system, but the code was not being exercised.

Comment (jd-lara) [reply src/subsystems.jl:243]: I wonder why this test system was working

File: src/base.jl [hunk 3]
```diff
@@ -274,11 +274,11 @@ function _post_deserialize_handling(sys::System; runchecks = true, assign_new_uu
     if assign_new_uuids
         IS.assign_new_uuid!(sys)
         for component in get_components(Component, sys)
-            assign_new_uuid!(sys, component)
+            IS.assign_new_uuid!(sys, component)
```
[the remark this answers is not in the document]
Comment (jd-lara) [reply src/base.jl:277]: This is ok, I don't think we should export this on PSY.

File: src/base.jl [hunk 4]
```diff
@@ -319,6 +319,60 @@ function System(sys_file::AbstractString, dyr_file::AbstractString; kwargs...)
...
 
+"""
+Construct a System from a subsystem of an existing system.
+"""
+function from_subsystem(sys::System, subsystem::AbstractString; runchecks = true)
+    if !in(subsystem, get_subsystems(sys))
+        error("subsystem = $subsystem is not stored")
+    end
+
+    # It would be faster to create an empty system and then populate it with
```
Comment (jd-lara) [inline src/base.jl:330]: The "real test" to this will be on the EI data set.

**Review by jd-lara — APPROVED** [review]

File: test/test_subsystems.jl [hunk 5]
```diff
@@ -70,7 +70,8 @@ function create_system_with_2_test_subsystems()
         subsystems[2],
         get_component(ACBus, c_sys5, "nodeC"),
     )
-
+    PSY.check(c_sys5)
+    PSY.check_components(c_sys5)
```
Comment (daniel-thom) [inline test/test_subsystems.jl:74]: The tests were passing because the checks were not being run.

**Pull request merged**